File: demo/Python/searchmode.py

import splunklib.client as client
import utils, time, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 3:
    print "Syntax: setmode.py \"search query\" [normal \ blocking]"
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
    
    if sys.argv[2] == "normal":     params = {"exec_mode" : "normal"}
    elif sys.argv[2] == "blocking": params = {"exec_mode" : "blocking"}
    else: raise Exception("unknown output mode")
    
    job = service.jobs.create(sys.argv[1], **params)
    
    if sys.argv[2] == "normal":
        sys.stdout.write("Job is running")
        sys.stdout.flush()
        while True:
            sys.stdout.write(".")
            sys.stdout.flush()
            time.sleep(1)
            job.refresh()
            if job.is_done():
                break    
    
    response = job.results()    
    print response

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: demo/Java/searchmode.java
import com.splunk.*; 
import com.splunk.JobArgs.ExecutionMode;

import java.util.*;
import java.io.*;

public class searchmode {

    public static void main(String[] args) {

		try {
			if(args.length != 2) {
				System.out.println("syntax: setmode \"search query\" [normal | blocking]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        JobArgs ja = new JobArgs();

	        if(args[1].equalsIgnoreCase("normal"))
	        		ja.setExecutionMode(ExecutionMode.NORMAL);
	        else if(args[1].equalsIgnoreCase("blocking"))
	        		ja.setExecutionMode(ExecutionMode.BLOCKING);
	        else 
	        	throw new Exception("Unknown mode " + args[1]);
	        	
			Job job = service.getJobs().create(args[0], ja);

	        if(args[1].equalsIgnoreCase("normal")) {
	        	System.out.print("Job is running");
	        	while(!job.isDone()) {
	        		System.out.print(".");
	        		Thread.sleep(1000);
	        	}
	        	System.out.println("complete!");
	        } 
	        InputStream response = job.getResults();
	        String line;	        
 			BufferedReader results = new BufferedReader(new InputStreamReader(response));
 			while ((line = results.readLine() ) != null) {
 				System.out.println(line);
 			}
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}